### PR TITLE
JBPM-8802 Backport SonarCloud fixes from kogito-runtimes

### PR DESCRIPTION
**Pull request by baldimir**

### CODE REVIEW

**Comment by baldimir** [comment]
jenkins execute full downstream build

**Comment by baldimir** [comment]
jenkins execute compile downstream build

**Comment by baldimir** [comment]
jenkins retest this

**Comment by baldimir** [comment]
jenkins retest this

**Comment by baldimir** [comment]
jenkins retest this

**Comment by baldimir** [comment]
jenkins retest this

File: jbpm-flow/src/main/java/org/jbpm/process/instance/event/listeners/RuleAwareProcessEventListener.java [hunk 1]
```diff
@@ -36,7 +35,7 @@
  * whenever process variable is updated.
  *
  */
-public class RuleAwareProcessEventLister implements ProcessEventListener {
+public class RuleAwareProcessEventListener implements ProcessEventListener {
```
[the remark this answers is not in the document]
Comment (baldimir) [reply jbpm-flow/src/main/java/org/jbpm/process/instance/event/listeners/RuleAwareProcessEventListener.java]: Oh ok, I will revert it back for now.

**Comment by baldimir** [comment]
jenkins execute full downstream build

**Comment by baldimir** [comment]
jenkins execute full downstream build

**Comment by MarianMacik** [comment]
jenkins retest this

**Comment by baldimir** [comment]
jenkins retest this

**Comment by baldimir** [comment]
jenkins retest this

**Comment by baldimir** [comment]
jenkins retest this

**Comment by baldimir** [comment]
jenkins execute full downstream build

**Comment by baldimir** [comment]
jenkins execute full downstream build

**Comment by baldimir** [comment]
jenkins execute compile downstream build

**Comment by baldimir** [comment]
jenkins execute compile downstream build

**Comment by baldimir** [comment]
jenkins execute compile downstream build

**Comment by baldimir** [comment]
jenkins execute full downstream build

**Comment by baldimir** [comment]
jenkins execute compile downstream build

**Comment by baldimir** [comment]
jenkins execute full downstream build

**Comment by baldimir** [comment]
jenkins execute full downstream build

**Comment by MarianMacik** [comment]
Downstream fails not related.

**Pull request merged**